common/ansi_filter.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common/ansi_filter.h"
+
+namespace cdc_ft {
+namespace ansi_filter {
+namespace {
+enum class State {
+  kNotInSequence,
+  kDCS,  // Starting with kESC + P or kDCSI, Device Control String.
+  kCS,   // Starting with kESC + [ or kCSI,  Control Sequence.
+  kOSC,  // Starting with kESC + ] or kOSCI, Operating System Command.
+};
+
+constexpr uint8_t kBEL = 0x07;   // Terminal bell.
+constexpr uint8_t kESC = 0x1B;   // ANSI escape character.
+constexpr uint8_t kST = 0x9C;    // String Terminator.
+constexpr uint8_t kDCSI = 0x90;  // Device Control String Introducer.
+constexpr uint8_t kCSI = 0x9B;   // Control Sequence Introducer.
+constexpr uint8_t kOSCI = 0x9D;  // Operating System Command Introducer
+
+}  // namespace
+
+std::string RemoveEscapeSequences(std::string str) {
+  size_t write_idx = 0;
+  State state = State::kNotInSequence;
+
+  for (size_t read_idx = 0; read_idx < str.size(); ++read_idx) {
+    uint8_t ch = static_cast<uint8_t>(str[read_idx]);
```

Comment:
In general, I would suggest to use fewer abbreviations (as suggested by the style guide)

Reply:
Done.

Reply:
I think this is still open.